modules/cache/main.tf

```diff
@@ -0,0 +1,66 @@
+data "aws_route53_zone" "public" {
+  name = var.zone_name
+}
+
+module "redis" {
+  count = var.cache_store == "redis" ? 1 : 0
+
+  source  = "cloudposse/elasticache-redis/aws"
+  version = "0.30.0"
+
+  name = "redis"
+
+  vpc_id               = var.vpc_id
+  namespace            = var.project
+  stage                = var.environment
+  subnets              = var.subnet_ids
+  availability_zones   = var.availability_zones
+  replication_group_id = "${var.project}-${var.environment}-redis"
+
+  family         = var.redis_family
+  engine_version = var.redis_engine_version
+  instance_type  = var.instance_type
+  cluster_size   = var.cluster_size
+
+  allowed_security_groups = var.security_groups
+
+  zone_id = data.aws_route53_zone.public.zone_id
```

Comment:
I don't think we should use this feature. We're not using the record it creates, and by putting it into the public zone we would be exposing internals of the user's system. Elasticache exposes the cluster through a single hostname anyway so this isn't necessary.

Reply:
yeah, make it fully secured.